src/test/java/com/answerdigital/answerking/service/ProductServiceTest.java

```diff
@@ -58,161 +62,326 @@ private ProductServiceTest() {
+        final Product product = productTestBuilder
+            .withDefaultValues()
+            .build();
+        final Product productTwo = productTestBuilder
+            .withDefaultValues()
+            .withId(2L)
+            .withName("Beans & Toast")
+            .build();
+        final Category category = categoryTestBuilder
+            .withDefaultValues()
+            .withProducts(Set.of(product, productTwo))
+            .build();
+
+        // when
+        doReturn(List.of(product, productTwo))
+            .when(productRepository)
+            .findProductsByCategoryId(anyLong());
+
+        final List<ProductResponse> response = productService.findProductsByCategoryId(category.getId());
+
+        // then
+        assertAll("All should be equal",
+            () -> assertEquals(2, response.size()),
+            () -> assertEquals(response.get(0).getName(), product.getName()),
+            () -> assertEquals(response.get(1).getName(), productTwo.getName())
+        );
+        verify(productRepository).findProductsByCategoryId(category.getId());
+    }
+
+    @Test
+    void testFindAllProductsInListOfIdsReturnsAListOfProducts() {
+        // given
+        final Product product = productTestBuilder
+            .withDefaultValues()
+            .build();
+        final Product productTwo = productTestBuilder
+            .withDefaultValues()
+            .withId(2L)
+            .build();
+
+        // when
+        doReturn(List.of(product, productTwo))
+            .when(productRepository)
+            .findAllByIdIn(anyList());
+
+        final List<Product> response = productService.findAllProductsInListOfIds(List.of(
+            product.getId(),
+            productTwo.getId()
+        ));
+
+        // then
+        assertAll("All should be equal",
+            () -> assertEquals(2, response.size()),
+            () -> assertEquals(response.get(0).getName(), product.getName()),
+            () -> assertEquals(response.get(1).getName(), productTwo.getName())
+        );
+        verify(productRepository).findAllByIdIn(anyList());
+    }
+
+    /**
+     * Helper method which asserts the equality of a Product against a ProductResponse object.
+     * @param product The Product to compare.
+     * @param response The ProductResponse to compare.
+     */
+    private void assertProductVsProductResponseEquality(final Product product, final ProductResponse response) {
+        assertAll("Product vs ProductResponse Equality",
+            () -> assertEquals(product.getId(), response.getId()),
+            () -> assertEquals(product.getName(), response.getName()),
+            () -> assertEquals(product.getDescription(), response.getDescription()),
+            () -> assertEquals(product.getPrice(), response.getPrice()),
+            () -> assertEquals(product.getCategory().getName(), response.getCategory().getName()),
+            () -> assertEquals(product.isRetired(), response.isRetired())
+        );
+    }
```

Comment:
Can anyone think of a better way of doing this, something doesn't sit quite right with me with the way I've did this.

Reply:
@AndyFlintAnswerDigital @oleksiiRybak @AudriusPutys @MohammadIqbalAD